pyproject.toml

```diff
@@ -43,8 +43,6 @@ globus-compute-sdk = "<2.28.0"
 tabulate = "^0.9.0"
 types-tabulate = "^0.9.0.3"
 nbformat = "^5.9.2"
-docker = "^6.1.3"
```

Comment:
I think we can clean more of these -- double check me on these but I don't think we need to depend on any of the following: 
- `dill` (probably safest as a transitive globus compute dependency now) 
- `gitpython` / `huggingface-hub` (no more modelconnectors) 
- `jinja2` / `nbformat` (no more notebooks)

Reply:
we can also remove/bump the ceiling on the globus compute sdk itself